clinica/pipelines/deeplearning_prepare_data/deeplearning_prepare_data_pipeline.py

```diff
@@ -138,15 +138,15 @@ def build_input_node(self):
                 self.caps_directory, "masks", f"tpl-{self.template}"
             )
 
-            if self.roi_list is None:
+            if not self.roi_list:
                 raise ValueError("A list of regions must be given.")
             else:
                 check_mask_list(
                     self.masks_location,
                     self.roi_list,
                     self.mask_pattern,
                     None
-                    if self.parameters.get("use_uncropped_image") is None
+                    if not self.parameters.get("use_uncropped_image")
                     else not self.parameters.get("use_uncropped_image"),
```

Comment:
This part of the code gets me very confused for the following reasons:

- Use of double negation: we are doing a boolean negation of a boolean flag named `use_uncropped_image`.
- The value computed is in fact not boolean but ternary since it also accepts `None`. This looks intended since further in the code path, in the `cropping` parameter of the `find_mask_path` function, there is a 3-path branching being done.

Besides the questionable sanity of the code, this won't work with Click. `use_uncropped_image` will always receive a value since it is declared as a flag. If we really need to distinguish between 3 cases, then we should be using an enum.

Reply:
Good catch on the `find_mask_path' three path branching. I updated the code to reflect the use of a boolean variable. I agree that the double negation can be confusing but I opted for minimal-changes-that-make-this-work for now as:
-The function is meant to migrate soon
- Input from @ravih18 would be valuable before refactoring

Reply:
Which means now this line can be changed to just `cropping=not self.parameters.get("use_uncropped_image")`

Reply:
Fixed